src/store/lib/useHostingStore.js

```diff
@@ -0,0 +1,234 @@
+import { useState, useCallback, useMemo } from 'react';
+import { Nothing } from 'folktale/maybe';
+import * as ob from 'urbit-ob';
+
+import { getVaneName, getVaneNumber } from 'lib/hosting';
+import SolarisClient from 'lib/SolarisClient';
+
+import { usePointCursor } from 'store/pointCursor';
+
+const STATE = {
+  UNKNOWN: 'UNKNOWN',
+  MISSING: 'MISSING',
+  RUNNING: 'RUNNING',
+  PENDING: 'PENDING',
+};
+
+function useHostingStore(url) {
+  const client = new SolarisClient(url);
+
+  const { pointCursor } = usePointCursor();
+
+  const [bootProgress, setBootProgress] = useState(0);
+  const [bootMessage, setBootMessage] = useState('Assembling Urbit');
+  const [startTime] = useState(Date.now());
+
+  const [status, _setStatus] = useState(STATE.UNKNOWN);
+  const setStatus = useCallback(
+    s => {
+      console.log(s);
+      _setStatus(s);
+    },
+    [_setStatus]
+  );
+
+  const [sysEvents, setSysEvents] = useState([]);
+  const [newEvents, setNewEvents] = useState([]);
+  const [runEvents, setRunEvents] = useState([]);
+
+  const [error, _setError] = useState();
+  const setError = useCallback(
+    error => {
+      _setError(error);
+      if (error) {
+        setStatus(STATE.UNKNOWN);
+        console.error(error);
+      }
+    },
+    [_setError, setStatus]
+  );
+
+  const unknown = status === STATE.UNKNOWN;
+  const missing = status === STATE.MISSING;
+  const running = status === STATE.RUNNING;
+  const pending = status === STATE.PENDING;
+
+  let syncStatus;
+
+  const resetEvents = useCallback(() => {
+    setSysEvents([]);
+    setNewEvents([]);
+    setRunEvents([]);
+  }, [setSysEvents, setNewEvents, setRunEvents]);
+
+  const getEvents = useCallback(() => {
+    try {
+      if (Nothing.hasInstance(pointCursor)) {
+        return;
+      }
+      const patp = ob.patp(pointCursor.value);
+      setError(undefined);
+
+      resetEvents();
+
+      const url = client.getShipsByPatpUrl(patp) + '/events';
+      const source = new EventSource(url);
+
+      // source.onerror = () => setStatus(STATE.UNKNOWN);
+      // source.onmessage = console.log;
```

Comment:
Is this debug tooling? Maybe only run it in development mode.
There's some commented out `console.log`s further down, too.